main.go

```diff
@@ -111,69 +185,117 @@ func checkShellyUpdateStatus(hostname string) (*shellyUpdateStatusResponse, erro
+		fmt.Printf("[%s,%s] failed to query update status: %s, aborting...\n", name, address, err)
 		return
 	}
 
 	// If there's an update available, trigger the update
-	if updateStatus.HasUpdate {
+	if (updateStage == "stable" && updateStatus.HasUpdate) ||
+		(updateStage == "beta" && updateStatus.OldVersion != updateStatus.BetaVersion) {
+		newVersion := updateStatus.NewVersion
+		if updateStage == "beta" {
+			newVersion = updateStatus.BetaVersion
+		}
 		fmt.Printf(
-			"[%s] update available! (%s -> %s), updating...\n",
-			instance.HostName, updateStatus.OldVersion,
-			updateStatus.NewVersion,
+			"[%s,%s] update available! (%s -> %s), updating...\n",
+			name, address, updateStatus.OldVersion, newVersion,
 		)
 
-		updateStatus, err := triggerShellyUpdate(shellyAddress)
+		updateStatus, err := triggerShellyUpdate(address)
 		if err != nil {
-			fmt.Printf("[%s] failed to start update: %s, aborting...\n", instance.Instance, err)
+			fmt.Printf("[%s,%s] failed to start update: %s, aborting...\n", name, address, err)
 			return
 		}
 
 		for updateStatus.Status == "updating" {
 			time.Sleep(time.Second * 5)
-			updateStatusCheck, err := checkShellyUpdateStatus(shellyAddress)
+			updateStatusCheck, err := checkShellyUpdateStatus(address)
 			if err != nil {
-				fmt.Printf("[%s] failed to query update status: %s, retrying...\n", instance.Instance, err)
+				fmt.Printf("[%s,%s] failed to query update status: %s, retrying...\n", name, address, err)
 			} else {
 				updateStatus = updateStatusCheck
 			}
 		}
 
-		fmt.Printf("[%s] device updated to %s!\n", instance.Instance, updateStatus.OldVersion)
+		fmt.Printf("[%s,%s] device updated to %s!\n", name, address, updateStatus.OldVersion)
 	} else {
-		fmt.Printf("[%s] already up to date (%s)\n", instance.Instance, updateStatus.OldVersion)
+		fmt.Printf("[%s,%s] already up to date (%s)\n", name, address, updateStatus.OldVersion)
+	}
+}
+
+func updateShellyGen2(name, address string) {
+	// First, we trigger a check for updates
+	fmt.Printf("[%s,%s] checking for updates (gen2)...\n", name, address)
+	updates, err := makeCheckForUpdateRequest(address)
+	if err != nil {
+		fmt.Printf("[%s,%s] failed to check for updates: %s, aborting...\n", name, address, err)
+		return
+	}
+
+	updateVersion := updates.Stable.Version
+	if updateStage == "beta" {
+		updateVersion = updates.Beta.Version
+	}
+	if updateVersion == "" {
+		fmt.Printf("[%s,%s] already up to date\n", name, address)
+		return
+	}
+
+	fmt.Printf("[%s,%s] updating to version %s...\n", name, address, updateVersion)
+	err = makeUpdateRequest(address, updateStage)
+	if err != nil {
+		fmt.Printf("[%s,%s] failed to update: %s, aborting...\n", name, address, err)
+		return
+	}
+}
+
+func updateShelly(name, address string, txtRecords []string, generation int) {
+	for _, txt := range txtRecords {
+		if txt == "gen=2" {
+			if generation == 2 || generation == 0 {
+				updateShellyGen2(name, address)
+			}
+			return
+		}
+	}
```

Comment:
This can be simplified:

```suggestion
	if slices.Contains(txtRecords, "gen=2") {
		if generation == 2 || generation == 0 {
			updateShellyGen2(name, address)
		}
		return
	}
```

Reply:
Thanks!